test/test_common.c

```diff
@@ -216,19 +206,38 @@ void setupRedisServer() {
     redisPort = findFreePort(6500, 6600);
 
     if (pid == 0) { /* child */
-        char redisPortStr[10];
-        char fullpath[256];
+        char redisPortStr[10], fullpath[256], testrdbModulePath[256];
 
-        printf ("Found free port to run Redis: %d\n", redisPort);
+        printf("Found free port to run Redis: %d\n", redisPort);
 
-        snprintf(fullpath, 255, "%s/%s", redisInstallFolder, "redis-server");
+        snprintf(fullpath, 255, "%s/redis-server", redisInstallFolder);
+        snprintf(testrdbModulePath, 255, "%s/../tests/modules/testrdb.so", redisInstallFolder);
         snprintf(redisPortStr, sizeof(redisPortStr), "%d", redisPort);
-        execl(fullpath, fullpath, "--port", redisPortStr , "--dir", "./test/tmp/", "--logfile", "./redis.log", (char*)NULL);
+
+        /* if module testrdb.so exists (ci.yaml takes care to build testrdb), part
+         * of redis repo testing, then load it for test_rdb_to_redis_module. The
+         * test will run only if testrdb appear in the server "MODULE LIST",
+         * otherwise skipped gracefully. */
+        if (access(testrdbModulePath, F_OK) != -1) {
+            execl(fullpath, fullpath,
+                  "--port", redisPortStr,
+                  "--dir", "./test/tmp/",
+                  "--logfile", "./redis.log",
+                  "--loadmodule", testrdbModulePath,
```

Comment:
do we have a test that uses RESTORE command and validates the module data was loaded on the dest correctly?

Reply:
See function `test_rdb_to_redis_module`. It might be little fragile though. As precondition, the test verifies that testrdb module is loaded into redis, otherwise the test gracefully skipped.

(Note that all quick and dirty `redis-cli` system calls should be replaced with integrtaing `hiredis` library eventually)

Reply:
great.. why skip it if the module load fails? i'm afraid we'll miss it instead of fixing the problem / improving the test.

Reply:
ci.yml should fail if failed to build `testrdb.so`. If we reach here and the file exists then for sure the test will capture the fail.
The graceful fallback to the case `testrdb.so` doesn't exists is to be able, later on, to run these tests against earliest redis version that I can reach. In addition there is the option to expose the tests for external redis server which can assist debugging Redis side in the future.

Reply:
it's a little odd to depend on the module from redis, and even odder that it's loosely based on it (graceful failures).
maybe we should bring that module (with some simplifications) into this repo and avoid that dependency.

Reply:
Note that compiling module requires to have redis source. I prefer to leave it as is quick and dirty (?!). And when we reach to a point that we want to expand the support in modules, then make the effort. 
Note that graceful failure is when the `testrdb.so` is not available. In our CI I will take care that it will verify success of `make -C tests/modules`

Reply:
compiling a module does NOT require the redis source, only redismodule.h, which is commonly just copied into the module's source tree.
i think your test better be self-sufficient and have their own module (copied from redis and cleaned).
you can leave that for a later step, but i think that's the right approach.

Reply:
I agree with you that this is the right approach... I am too lazy about it because I know we goanna  expand it and write it properly (It can be valuable feature for modules)